### PR TITLE
Avoid haskell and jvm bazel blackduck scan stomping

### PR DESCRIPTION
**Pull request by dasormeter**
While cross-checking against Gary's PRs cleaning up license cruft, I noticed that haskell libs were not in the NOTICES file.

This is because the jvm bazel scan run after the haskell scan stomps the results since the scans share a name.

This PR gives a unique code location to haskell vs jvm scan so they will not stomp on each other, and the final scan results and NOTICES file generated at the end is comprehensive.

Also makes sure the scan is run with the actual branch name, and not always with `master` regardless of where it is run from

### CODE REVIEW

**Comment by dasormeter** [comment]
superceded by https://github.com/digital-asset/daml/pull/8247

**Review by garyverhaegen-da — APPROVED** [review]

**Review by cocreature — APPROVED** [review]

File: ci/cron/daily-compat.yml [hunk 1]
```diff
@@ -240,7 +242,7 @@ jobs:
           set -euo pipefail
           eval "$(./dev-env/bin/dade-assist)"
           bash <(curl -s https://raw.githubusercontent.com/DACH-NY/security-blackduck/master/synopsys-detect) \
-          ci-build digital-asset_daml master \
+          ci-build digital-asset_daml $Build.SourceBranchName) \
```
Comment (garyverhaegen-da) [inline ci/cron/daily-compat.yml:245]:
```suggestion
          ci-build digital-asset_daml $(Build.SourceBranchName) \
```

**Comment by dasormeter** [comment]
Got a clean scan 
image

haskell libs are there
image


jvm libs are there
image

**Pull request merged**